src/main/java/hashtable/HashTable2.java

```diff
@@ -0,0 +1,90 @@
+package hashtable;
+
+public class HashTable2 {
+    private Item[] table;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private int count;
+
+    public HashTable2(int size) {
+        if(size <= 0) {
+            throw new IllegalArgumentException("The size of table must be more than 0!");
+        }
+        table = new Item[size];
+        for (int i = 0; i < size; i++) {
+            table[i] = null;
+        }
+    }
+
+    public void put(Object key, Object value) {
+        Item item = new Item(key, value);
+        int idx = findIdxToInsert(key);
+        table[idx] = item;
+        count++;
+        double loadFactor = ((double)count) / table.length;
+        if (loadFactor > DEFAULT_LOAD_FACTOR) {
+            rehash();
+        }
+    }
+
+    public void remove(Object key) {
+        int hash = findIdxToInsert(key);
+        if(table[hash] != null && table[hash] != Item.DELETED) {
+            table[hash] = Item.DELETED;
+        }
+    }
+
+    public Object get(Object key) {
+        if (key == null) {
+            return null;
+        }
+        int hash = findIdxToInsert(key);
+        if (table[hash] == null || table[hash] == Item.DELETED) {
+            return 0;
+        } else {
+            return table[hash].getValue();
+        }
+    }
+
+    int getSize() {
+        return table.length;
```

Comment:
did you mean to return `count` here?

Reply:
No, I need the length of array in order to verify that resizing works correctly.

Reply:
In real projects you wouldn't test internal behaviour of classes. It's important that its interface (public API) works, and how it works - it's not what tests usually check.